docs/DebuggerHTML-Components-Reducers-Actions.md

```diff
@@ -0,0 +1,663 @@
+
+* The Breakpoints component renders a list of all existing breakpoints.
+
+* The Frames component is responsible for rendering the current call stack when a breakpoint is reached.
+
+* The Scopes component is responsible for rendering the current variable scopes for the given breakpoint. It uses the ObjectInpector component to render the tree for all scopes and variables. The state of which nodes are collapsed/expanded are maintained using a ManagedTree component, in similar fashion to the SourcesTree component.
+
+![](https://docs.google.com/drawings/d/1zHogPebNmOFT9Xx6cZsaA6R6cTQLUzBXePV9sf62chA/pub?w=960&h=720)
+[Click here to Edit](https://docs.google.com/drawings/d/1zHogPebNmOFT9Xx6cZsaA6R6cTQLUzBXePV9sf62chA/edit?usp=sharing)
+
+#Component Data
+
+Some components in Debugger.html are aware of the Redux store; others are
+not and are just rendering passed in properties. The Redux-aware
+components are connected to the Redux store using the <code>connect()</code> method, as illustrated by the following code:
+
+<pre>javascript
+const React = require("react");
+const { connect } = require("react-redux");
+const { bindActionCreators } = require("redux");
+.
+.
+const actions = require("../actions");
+.
+.
+.
+
+module.exports = connect(
+state =&gt; ({ pauseInfo: getPause(state),
+expressions: getExpressions(state) }),
+dispatch => bindActionCreators(actions, dispatch)
+)(Expressions);</pre>
+
+This example shows the Expressions component, which should be aware of
+the Redux state. We are using Redux’s <code>connect()</code> method to connect to the
+Redux store. This example is pulling in <code>pauseInfo</code> and <code>Expressions</code> from
+the Redux state. Finally, all of the actions in the actions folder are
+combined and the contained <code>actionCreators</code> in each of the files are setup
+so the actions can be called directly from the component.
+
+#Reducers
+
+
+The Reducers are all located in the public/js/reducers folder and are
+all combined using Redux’s <code>combineReducters()</code> function. This function is
+executed in main.js as follows:
+
+<pre>javascript
+const reducers = require("./reducers");
+
+.
+
+.
+
+const store = createStore(combineReducers(reducers));</pre>
+
+All of the reducers are combined using the index.js file in the
+reducers folder. In the Debbuger.html project, each reducer has an
+<code>update()</code> function to handle actions for its slice of state.
+
+##Asynch-requests
+
+The **asynch-requests** reducer creates an array that stores a unique
+sequence number for every promise being executed from an action.
+It removes the sequence number from the array when a specific promise
+resolves or rejects. The following image shows a snapshot of the
+debugger.html state with an active promise.
+
+![](images/asynchreducer.png)
+
+##Breakpoints
+
+The **breakpoints reducer** is responsible for handling breakpoint
+state. It adds an Immutable Map of breakpoint objects and a Boolean

```

Comment:
I'd like to see if we can avoid screenshots.

Do you think we could have a markdown table for the fields, types, descriptions?
Also, maybe include an example data structure so for the case of a Frame - that is vague, but an example might make it clear.